poetry/masonry/builders/sdist.py

```diff
@@ -64,7 +64,7 @@ def build(self, target_dir=None):  # type: (Path) -> Path
         target = target_dir / "{}-{}.tar.gz".format(
             self._package.pretty_name, self._meta.version
         )
-        gz = GzipFile(target.as_posix(), mode="wb")
+        gz = GzipFile(target.as_posix(), mode="wb", mtime=0)
```

Comment:
Hello @achow101,

thanks a lot for your contribution. Fixing this looks necessary to me. Are there any reasons for settings the mtime to 0 and not to the current time via `time.time()`?

fin swimmer

Reply:
It's literally the whole point of this PR. A fixed time is needed otherwise the resulting archive is non-deterministic. Setting to `time.time()` means that the timestamp will continue to be variable which is what I'm trying to fix.

Reply:
Ah , sorry it was obviously to late yesterday. The current timestamp is already used if the parameter is omitted and I thought it is something similar to the already merged #1541.

So let's change my question :) Why do you want the file to be deterministic?

Reply:
I subscribe to the idea that the releases that you publish should be built in such a way that other people can repeat the build process and produce the exact same files, down to the bit. This makes it much easier to audit the published binaries/archives. https://reproducible-builds.org/ basically explains why.

There's also a bunch of things already in poetry which make the build results more reproducible, so reproducibility is clearly something that is intended in this project.